rust/operator-binary/src/crd/security.rs

```diff
@@ -672,7 +681,7 @@ impl KafkaTlsSecurity {
             "-X".to_string(),
             format!("sasl.kerberos.service.name={service_name}"),
             "-X".to_string(),
-            format!("sasl.kerberos.principal={service_name}/{pod_fqdn}@$KERBEROS_REALM"),
+            format!("sasl.kerberos.principal={service_name}/{pod_kcat}@$KERBEROS_REALM"),
```

Comment:
`pod_kcat` contains the Listener service name, e.g. `kafka-broker-default-0-listener-broker.stackable-products.svc.cluster.local`. This works for the broker listener class `cluster-internal`. But if the listener class is set to `external-unstable`, then the Keytab contains only the Node IP address:

```
KVNO Timestamp         Principal
---- ----------------- --------------------------------------------------------
   1 03/14/25 11:06:28 kafka/172.18.0.2@SBLE.TEST
```

Reply:
I've changed the test to run all 3 bootstrap listeners for 1 (cluster-internal) broker listener.